### ISSUE TITLE
Capitalized properties break generated classes

### ISSUE THREAD

**Issue by haasted** [immutables/immutables#290]
I tried updating from 2.1.0 to version 2.1.15 of the framework, but encountered a strange issue. It seems that properties starting with capital letters are not included in the generated classes.

The following code reproduces the issue:
```
@Value.Immutable
public abstract class Foo {
    abstract String Name();
}
```
The `ImmutableFoo` class simply does not override the `Name`-method.

Changing "Name" to "name" fixes the issue, and a `name()` method is included in `ImmutableFoo` after re-compiling.

I played around with the Maven version, and it seems that changing to 2.1.11 fixes the issue, while 2.1.13 contains it.

**Comment by elucash** [comment]
```
src/bar/Foo.java:7: warning: Attributes with names starting with uppercase letters are not supported and are ignored. Sorry.
    abstract String Name();
                    ^
1 warning
```
We have a warning for this, however it might not be always reported. While it was kinda allowed previously, we had wrong java code generated due to shortcomings of our rewrite imports processing, it was confused method names for fully qualified class names in a lot of cases. Could you use corresponding annotation to give serialized name. Or in case of Gson on JVM, you may also use field naming strategy support.
There is also a way to actually fix this by disabling import post-processing if we detect any such capitalized name, but in this case generated code would be more verbose/ugly.

**Comment by elucash** [comment]
Should also detect capitalization in packages as mentioned in #295

**Comment by haasted** [comment]
Thanks for the quick response. I must admit to have overlooked the very informative warning. Can verify that it was shown, though.

**Comment by elucash** [comment]
Released in v 2.1.16 Thank you!
